PWGHF/HFL/Tasks/taskHfeSelection.cxx

```diff
@@ -0,0 +1,284 @@
+
+      for (const auto& ematchTrack : tracksofcluster) {
+
+        auto matchTrack = ematchTrack.template track_as<TracksType>();
+
+        auto cluster = ematchTrack.template emcalcluster_as<ClusterType>();
+
+        phiMatchTrack = matchTrack.phi();
+        etaMatchTrack = matchTrack.eta();
+        pMatchTrack = matchTrack.p();
+        ptMatchTrack = matchTrack.pt();
+        tpcNsigmaMatchTrack = matchTrack.tpcNSigmaEl();
+        phiMatchCluster = cluster.phi();
+        etaMatchCluster = cluster.eta();
+        eMatchCluster = cluster.energy();
+        m02MatchCluster = cluster.m02();
+        m20MatchCluster = cluster.m20();
+        timeCluster = cluster.time();
+        cellCluster = cluster.nCells();
+
+        deltaPhiMatch = matchTrack.trackPhiEmcal() - phiMatchCluster;
+        deltaEtaMatch = matchTrack.trackEtaEmcal() - etaMatchCluster;
+
+        registry.fill(HIST("hClsTrkEtaPhiDiffTime"), deltaEtaMatch, deltaPhiMatch, cluster.time());
+        // Track and cluster Matching
+        if (std::abs(deltaPhiMatch) > deltaPhiMatchMin || std::abs(deltaEtaMatch) > deltaEtaMatchMin)
+          continue;
+        if (std::abs(timeCluster) > clusterTimeMax)
+          continue;
+        if (fillClusterInfo)
+          registry.fill(HIST("hClusterInformationAfter"), eMatchCluster, etaMatchCluster, phiMatchCluster, cluster.nCells(), timeCluster);
+        float eop = eMatchCluster / pMatchTrack;
+        registry.fill(HIST("hPIDafterMatch"), eop, matchTrack.tpcSignal(), tpcNsigmaMatchTrack, pMatchTrack, ptMatchTrack, etaMatchTrack, phiMatchTrack);
+
+        // Apply Electron Identification cuts
+
+        if ((tpcNsigmaMatchTrack < tpcNsigmaElectronMin || tpcNsigmaMatchTrack > tpcNsigmaElectronMax) || (m02MatchCluster < m02ElectronMin || m02MatchCluster > m02ElectronMax) || (m20MatchCluster < m20ElectronMin || m20MatchCluster > m20ElectronMax))
+          continue;
+        registry.fill(HIST("hEPRatioafterPID"), eop, ptMatchTrack);
+        if (eop < eopElectronMin || eop > eopElectronMax)
+          continue;
+
+        registry.fill(HIST("hPIDafterPIDcuts"), eop, pMatchTrack, ptMatchTrack, tpcNsigmaMatchTrack, eMatchCluster, m02MatchCluster, m20MatchCluster);
+        isEMcal = 1;
+        electronSel(matchTrack.collisionId(), matchTrack.globalIndex(), matchTrack.eta(), matchTrack.phi(), ptMatchTrack, pMatchTrack, rapidity, matchTrack.dcaXY(), matchTrack.dcaZ(), matchTrack.tpcNSigmaEl(), matchTrack.tofNSigmaEl(),
+                    eMatchCluster, etaMatchCluster, phiMatchCluster, m02MatchCluster, m20MatchCluster, cellCluster, timeCluster, deltaEtaMatch, deltaPhiMatch, isEMcal);
+      }
+
+      /// Electron information without Emcal and use TPC and TOF
+      if (isEMcal == 1)
+        continue;
+      electronSel(track.collisionId(), track.globalIndex(), etaTrack, phiTrack, ptTrack, pTrack, rapidity, dcaxyTrack, dcazTrack, track.tpcNSigmaEl(), track.tofNSigmaEl(),
```

Comment:
why not fill the electron table in a different place inside PWGHF
HFL/TableProducer and then have a task applying electron cuts and obtaining hadron contamination in the Tasks folder? If you do this, you should apply loose cuts for filling the table, so that hadron contamination can also be estimated from the table.

Reply:
Okay, I will fill the electron table inside PWGHF/HFL/TableProducer.